app/uk/gov/hmrc/economiccrimelevyregistration/controllers/CheckYourAnswersController.scala

```diff
@@ -101,7 +101,13 @@ class CheckYourAnswersController @Inject() (
       _         = emailService.sendRegistrationSubmittedEmails(request.registration.contacts, response.eclReference)
       _        <- eclRegistrationConnector.deleteRegistration(request.internalId)
     } yield Redirect(routes.RegistrationSubmittedController.onPageLoad()).withSession(
-      request.session + (SessionKeys.EclReference -> response.eclReference)
+      request.session ++ Seq(
+        SessionKeys.EclReference              -> response.eclReference,
+        SessionKeys.FirstContactEmailAddress  -> request.registration.contacts.firstContactDetails.emailAddress
+          .getOrElse(throw new IllegalStateException("First contact email address not found in registration data")),
+        SessionKeys.SecondContactEmailAddress -> request.registration.contacts.secondContactDetails.emailAddress
+          .getOrElse("")
+      )
```

Comment:
sorry to be a pain with more comments, but I think we could take this a step further, see below

we can just make it so we only populate the session with the email if it's there, this makes the code simpler on the `RegistrationSubmittedController`

```suggestion
      val updatedSession = request.session ++ Seq(
        SessionKeys.EclReference             -> response.eclReference,
        SessionKeys.FirstContactEmailAddress -> request.registration.contacts.firstContactDetails.emailAddress
          .getOrElse(throw new IllegalStateException("First contact email address not found in registration data"))
      )

      Redirect(routes.RegistrationSubmittedController.onPageLoad()).withSession(
        request.registration.contacts.secondContactDetails.emailAddress.fold(updatedSession)(email =>
          updatedSession ++ Seq(SessionKeys.SecondContactEmailAddress -> email)
        )
      )
```